Sources/Controllers/2.0/OARearrangeCustomFiltersViewController.mm

```diff
@@ -0,0 +1,453 @@
+    if (indexPath.section == kAllFiltersSection)
+        [self hideMode:indexPath];
+    else if (indexPath.section == kHiddenFiltersSection)
+        [self restoreMode:indexPath];
+}
+
+- (void)hideMode:(NSIndexPath *)indexPath
+{
+    OAEditFilterItem *filterItem = _filtersItems[indexPath.row];
+    [_filtersItems removeObject:filterItem];
+    [_hiddenFiltersItems addObject:filterItem];
+    [_hiddenFiltersKeys addObject:filterItem.filter.filterId];
+    filterItem.filter.isActive = NO;
+    [self updateFiltersIndexes];
+    NSIndexPath *targetPath = [NSIndexPath indexPathForRow:_hiddenFiltersItems.count - 1 inSection:kHiddenFiltersSection];
+    [CATransaction begin];
+    [CATransaction setCompletionBlock:^{
+        [_tableView reloadData];
+    }];
+    [_tableView beginUpdates];
+    [_tableView moveRowAtIndexPath:indexPath toIndexPath:targetPath];
+    [_tableView endUpdates];
+    [CATransaction commit];
+}
+
+- (void)restoreMode:(NSIndexPath *)indexPath
+{
+    OAEditFilterItem *filterItem = _hiddenFiltersItems[indexPath.row];
+    int order = filterItem.order;
+    order = order > _filtersItems.count ? (int) _filtersItems.count : order;
+    NSIndexPath *targetPath = [NSIndexPath indexPathForRow:order inSection:kAllFiltersSection];
+    [CATransaction begin];
+    [CATransaction setCompletionBlock:^{
+        [_tableView reloadData];
+    }];
+    [_hiddenFiltersItems removeObjectAtIndex:indexPath.row];
+    [_filtersItems insertObject:filterItem atIndex:order];
+    [_hiddenFiltersKeys removeObject:filterItem.filter.filterId];
+    filterItem.filter.isActive = YES;
+    [_tableView beginUpdates];
+    [_tableView moveRowAtIndexPath:indexPath toIndexPath:targetPath];
+    [_tableView endUpdates];
+    [CATransaction commit];
+}
+
+- (void)updateFiltersIndexes
+{
+    for (int i = 0; i < _filtersItems.count; i++)
+        _filtersItems[i].order = i;
+}
+
+- (void)showChangesAlert
+{
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil message:OALocalizedString(@"osm_editing_lost_changes_title") preferredStyle:UIAlertControllerStyleAlert];
+    [alert addAction:[UIAlertAction actionWithTitle:OALocalizedString(@"shared_string_exit") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        [self.navigationController popViewControllerAnimated:YES];
+    }]];
+    [alert addAction:[UIAlertAction actionWithTitle:OALocalizedString(@"shared_string_cancel") style:UIAlertActionStyleCancel handler:nil]];
+    [self presentViewController:alert animated:YES completion:nil];
+}
+
+#pragma mark - UITableViewDataSource
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    NSString *cellType = indexPath.section == kActionsSection ? kButtonRightIconCell : kHideButtonCell;
+    if ([cellType isEqualToString:kHideButtonCell])
+    {
+        static NSString* const identifierCell = kHideButtonCell;
```

Comment:
There is no specific need to redefine the constant key ```kHideButtonCell``` here. You can use it directly instead of the ```identifierCell```